deployment-server/src/Deploy/Worker.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+module Deploy.Worker where
+
+import           Control.Concurrent.Chan     (Chan, readChan)
+import           Control.Monad               (forever)
+import           Control.Monad.IO.Class      (liftIO)
+import           Control.Monad.Reader        (runReaderT)
+import           Control.Newtype.Generics    (Newtype, unpack)
+import qualified Data.ByteString.Lazy.Char8  as BS
+import           Data.Char                   (isSpace)
+import           Data.Text                   (Text)
+import qualified Data.Text                   as Text
+import           GHC.Generics                (Generic)
+import           GitHub.Data.Webhooks.Events (PullRequestEvent)
+import           Options.Generic             (ParseField, ParseFields, ParseRecord, getRecord)
+import           System.Exit                 (ExitCode (ExitFailure, ExitSuccess))
+import           System.IO.Temp              (withSystemTempDirectory)
+import           System.Process.Typed        (readProcess, runProcess, setEnv, setWorkingDir, shell)
+import           Web.Slack                   (SlackConfig, chatPostMessage)
+import           Web.Slack.Chat              (mkPostMsgReq)
+
+newtype SlackChannel = SlackChannel Text
+    deriving stock (Generic)
+    deriving anyclass (Newtype, ParseFields, ParseRecord, ParseField)
+    deriving newtype (Show)
```

Comment:
Interesting. Does `deriving newtype (Show)` just show the inner text then, without prefixing it with the type name? I'll have to remember that one. :-)

Reply:
actually I'm not sure, this is left over code from when I had `Read` in here. The issue with `Read` in `newtype` is that it doesn't unwrap the `Text` so I had to `read "\"something\""`

Reply:
```
*Main Deploy.Server Deploy.Worker> show (read "some" :: SlackChannel)
"\"some\""
```
I should probably fix that

Reply:
done